web/modules/custom/og_addition/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,303 @@
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array.
+   */
+  public function buildFull(array $build, NodeInterface $entity): array {
+
+    $elements = [];
+
+    // Header.
+    $header = $this->buildHeader($entity);
+    $elements[] = $this->wrapContainerWide($header);
+
+    // Body.
+    $body = $this->buildProcessedText($entity, 'body');
+    $elements[] = $this->wrapContainerWide($body);
+
+    // Compute og message/links/text.
+    $elements[] = $this->buildOgMarkup($entity);
+
+    $build[] = $this->wrapContainerBottomPadding($elements);
+
+    return $build;
+  }
+
+  /**
+   * Build the header.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \IntlException
+   */
+  protected function buildHeader(NodeInterface $entity): array {
+    $elements = [];
+
+    $elements[] = $this->buildConditionalPageTitle($entity);
+
+    // Show the node type as a label.
+    $node_type = NodeType::load($entity->bundle());
+    $elements[] = $this->buildLabelsFromText([$node_type->label()]);
+
+    // Date.
+    $timestamp = $this->getFieldOrCreatedTimestamp($entity, 'field_publish_date');
+    $element = IntlDate::formatPattern($timestamp, 'long');
+
+    // Make text bigger.
+    $elements[] = $this->wrapTextDecorations($element, FALSE, FALSE, 'lg');
+
+    $elements = $this->wrapContainerVerticalSpacing($elements);
+    return $this->wrapContainerNarrow($elements);
+  }
+
+  /**
+   * Builds the OG output.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function buildOgMarkup(NodeInterface $entity) {
+    $user = $this->currentUser;
+
+    // Auto group owners/creator.
+    if ($entity->getOwnerId() == $user->id()) {
+      $og_item = [
+        '#type' => 'html_tag',
+        '#tag' => 'span',
+        '#attributes' => [
+          'title' => $this->t('You are the group manager'),
+          'class' => ['group', 'manager'],
+        ],
+        '#value' => $this->t('You are the group manager'),
+      ];
+
+      return $this->wrapContainerWide($og_item);
+    }
+    $storage = $this->entityTypeManager->getStorage('og_membership');
+    $props = [
+      'uid' => $user ? $user->id() : 0,
+      'entity_type' => $entity->getEntityTypeId(),
+      'entity_bundle' => $entity->bundle(),
+      'entity_id' => $entity->id(),
+    ];
+
+    $options = [];
+    $attr_class = [];
+    $text = "";
+    $group_name = $entity->label();
+    $name = $user->getDisplayName();
+    $parameters = [
+      'entity_type_id' => $entity->getEntityTypeId(),
+      'group' => $entity->id(),
+    ];
+
+    $memberships = $storage->loadByProperties($props);
```

Comment:
You are not using OG'a API here. Try to find the right solution..

Reply:
Yeah, I have replaced that implementation with the right one (og.membership_manager service).